src/java/htsjdk/samtools/SAMFileWriterImpl.java

```diff
@@ -171,9 +171,6 @@ private SAMRecordComparator makeComparator() {
     public void addAlignment(final SAMRecord alignment)
     {
         if (sortOrder.equals(SAMFileHeader.SortOrder.unsorted)) {
-            if (!header.getGroupOrder().equals(SAMFileHeader.GroupOrder.none)) {
-                throw new UnsupportedOperationException("GroupOrder " + header.getGroupOrder() + " is not supported");
-            }
```

Comment:
I have no idea why that check was there in the first place, and was rather surprised to find my code (a separate program in Picard) failing when I tried to create a SAMFileWriter with `SO=unsorted` and `GO=query`!  I went and read the spec and found absolutely no support for this assertion, and in fact found the opposite:

    Grouping of alignments, indicating that similar alignment records are grouped together but the
file is not necessarily sorted overall

I think the reality is that `GO` is very rarely used in reality, and has never really been used in HTSJDK or Picard, and that this check was implemented many years ago based on an incorrect understanding.